### DISCUSSION TITLE
Prediction on Training Data

### DISCUSSION THREAD

**Discussion by mahnoor-byte** [AammarTufail/python_ka_chilla2023#76] [Q&A]
1. Prediction values are different each time the code is executed. Is this how it is supposed to be? Attaching images below.

![image](https://user-images.githubusercontent.com/121869407/213370542-e47382ce-9a3f-45a0-bb75-4e02084e2f16.png)

![image](https://user-images.githubusercontent.com/121869407/213370596-31adfdd1-71d6-452c-bbd6-0d05b12fa863.png)

3. RMSE is not giving correct value while using:
`RMSE = mean_squared_error(y_true=y_test, y_pred=predictions, squared=False)`

![image](https://user-images.githubusercontent.com/121869407/213371318-9996e3eb-a366-4195-bbce-7f4fdd01c5ff.png)

**Comment by Muhammad-Ali-Butt** [comment]
@seharfatimaa baji aap ne rmse ko jab print kia tu variable R2 dia howa ha matlab likha howa ha **print('RMSE:', R2)** comma k baad RMSE likhay kyu k aap ne variable RMSE ko root mean squared assign kia ha hope this explains , basically galat variable assign kr dia print mai


edit : i forgot to answer your first question , i think its because of train test split kyu k ye function jo ha randomly split krta ha data ko so each time you run the code you get different result. this might explain why the result/output is different each time . im not sure tho so anyone reading this who knows the exact reason or if my judgement is right or not , kindly do verify here tagging me as well so i know if i was right or wrong.

**Reply by mahnoor-byte** [reply]
Thank you.

**Reply by Muhammad-Ali-Butt** [reply]
np i have edited my comment to explain your initial question as well btw im not 100% sure if thats the actual explanation so if anyone verifies then  good if not then you can confirm it by asking @AammarTufail during meeting time
